crates/solidity/outputs/cargo/crate/src/backend/l1/generated/nodes.rs

```diff
@@ -0,0 +1,1603 @@
+    pub node_id: usize,
+    pub end: Option<Expression>,
+}
+
+pub type PositionalArgumentsDeclaration = Rc<PositionalArgumentsDeclarationStruct>;
+
+#[derive(Debug)]
+pub struct PositionalArgumentsDeclarationStruct {
+    pub node_id: usize,
+    pub arguments: PositionalArguments,
+}
+
+pub type NamedArgumentsDeclaration = Rc<NamedArgumentsDeclarationStruct>;
+
+#[derive(Debug)]
+pub struct NamedArgumentsDeclarationStruct {
+    pub node_id: usize,
+    pub arguments: Option<NamedArgumentGroup>,
+}
+
+pub type NamedArgumentGroup = Rc<NamedArgumentGroupStruct>;
+
+#[derive(Debug)]
+pub struct NamedArgumentGroupStruct {
+    pub node_id: usize,
+    pub arguments: NamedArguments,
+}
+
+pub type NamedArgument = Rc<NamedArgumentStruct>;
+
+#[derive(Debug)]
+pub struct NamedArgumentStruct {
+    pub node_id: usize,
+    pub name: Rc<TerminalNode>,
+    pub value: Expression,
+}
+
+pub type TypeExpression = Rc<TypeExpressionStruct>;
+
+#[derive(Debug)]
+pub struct TypeExpressionStruct {
+    pub node_id: usize,
+    pub type_name: TypeName,
+}
+
+pub type NewExpression = Rc<NewExpressionStruct>;
+
+#[derive(Debug)]
+pub struct NewExpressionStruct {
+    pub node_id: usize,
+    pub type_name: TypeName,
+}
+
+pub type TupleExpression = Rc<TupleExpressionStruct>;
+
+#[derive(Debug)]
+pub struct TupleExpressionStruct {
+    pub node_id: usize,
+    pub items: TupleValues,
+}
+
+pub type TupleValue = Rc<TupleValueStruct>;
+
+#[derive(Debug)]
+pub struct TupleValueStruct {
+    pub node_id: usize,
+    pub expression: Option<Expression>,
+}
+
+pub type ArrayExpression = Rc<ArrayExpressionStruct>;
+
+#[derive(Debug)]
+pub struct ArrayExpressionStruct {
+    pub node_id: usize,
+    pub items: ArrayValues,
+}
+
+pub type HexNumberExpression = Rc<HexNumberExpressionStruct>;
+
+#[derive(Debug)]
+pub struct HexNumberExpressionStruct {
+    pub node_id: usize,
+    pub literal: Rc<TerminalNode>,
+}
+
+pub type DecimalNumberExpression = Rc<DecimalNumberExpressionStruct>;
+
+#[derive(Debug)]
+pub struct DecimalNumberExpressionStruct {
+    pub node_id: usize,
+    pub literal: Rc<TerminalNode>,
+    pub unit: Option<NumberUnit>,
+}
+
+pub type YulBlock = Rc<YulBlockStruct>;
+
+#[derive(Debug)]
+pub struct YulBlockStruct {
+    pub node_id: usize,
+    pub statements: YulStatements,
+}
+
+pub type YulFunctionDefinition = Rc<YulFunctionDefinitionStruct>;
+
+#[derive(Debug)]
+pub struct YulFunctionDefinitionStruct {
+    pub node_id: usize,
+    pub name: Rc<TerminalNode>,
+    pub parameters: YulParametersDeclaration,
+    pub returns: Option<YulReturnsDeclaration>,
+    pub body: YulBlock,
+}
+
+pub type YulParametersDeclaration = Rc<YulParametersDeclarationStruct>;
+
+#[derive(Debug)]
+pub struct YulParametersDeclarationStruct {
+    pub node_id: usize,
+    pub parameters: YulParameters,
+}
+
+pub type YulReturnsDeclaration = Rc<YulReturnsDeclarationStruct>;
+
+#[derive(Debug)]
+pub struct YulReturnsDeclarationStruct {
+    pub node_id: usize,
```

Comment:
I wonder why is this a raw node ID rather than the node RC directly?

```suggestion
    pub cst: Node,
```

Reply:
I'm on the fence with this. Saving the CST's `Rc` is convenient to navigate back to the CST on demand. But it makes the default `Debug` derived implementation close to useless, because it will recursively print the node and all its children. Using a raw `usize` (or `NodeId`) means we can easily render the IR tree for debugging purposes. Alternatively, we'd need to define custom implementations for `Debug` for each of these IR nodes.

Reply:
I'll add this to discuss on #1312